src/main/habit-modal-dialogue/habit-modal-dialogue.component.ts

```diff
@@ -0,0 +1,211 @@
+import { Component, ElementRef, OnInit, ViewChild } from '@angular/core';
+import { DynamicDialogRef } from "primeng/dynamicdialog";
+import { AppConstants, daysOfWeek, iconMap, TimeOfDay } from "../Constants/app-constant";
+import { DisplayService } from "../Service/display.service";
+import { HabitService } from "../Service/habit.service";
+import { ConfirmationService } from "primeng/api";
+
+@Component({
+  selector: 'app-habit-modal-dialogue',
+  templateUrl: './habit-modal-dialogue.component.html',
+  styleUrls: ['./habit-modal-dialogue.component.scss']
+})
+export class HabitModalDialogueComponent implements OnInit {
+
+  @ViewChild('calender') calenderDialogue!: ElementRef;
+  habitName!: string;
+  goal!: number;
+  frequency!: string;
+  goalValue: number = 1;
+  runningValue = 3;
```

Comment:
please verify if we are using above variables

Reply:
These variables are created to show initial values for different habits